core/src/main/java/org/testcontainers/utility/MountableFile.java

```diff
@@ -33,7 +37,11 @@
 @Slf4j
 public class MountableFile implements Transferable {
 
+    private static final int BASE_FILE_MODE = 0100000;
+    private static final int BASE_DIR_MODE = 0040000;
+
     private final String path;
+    private final int permissions;
```

Comment:
I first made it Integer, with default value null, but 0-check still looks better than null-check. Wdyt?

Reply:
Makes sense. I am still for int type. Check for -1 may be?

Reply:
Go for it. Since the primitive nature doesn't leak outside of the class, I think it's reasonable.